### ISSUE TITLE
Companion app extremely slow on Iphone

### ISSUE THREAD

**Issue by Solarflor** [home-assistant/frontend#20728]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

My wife have Iphone 13 and I have iPhone 15. Both phone’s companion app works extremely slow. It takes 15-20 seconds to register button clicks on dashboards or just moving through it .
Tried clearing front end cache. Tried reset. Tried reinstall. Same result
If I open HA with Safari I have the same problem. No proplem with Chrome. 

### Describe the behavior you expected

The problem started with previous version of HA version but now it is impossible to use it.
Also other users are complaining the same problem with Iphone

### Steps to reproduce the issue

1.The probelm appear in the main menu, Panoramic and other Dashboards 
2.
3.
...

### What version of Home Assistant Core has the issue?

2024.5.1

**Comment by piotrowskimarek** [comment]
Seeing same issue on my side, chrome is fine , safari (ios/macos) is really slow

**Comment by themistymay** [comment]
Adding a +1 to this and attaching some logs from an impacted iPhone.
[ha_logs.zip](https://github.com/home-assistant/frontend/files/15262863/ha_logs.zip)

**Comment by themistymay** [comment]
I can also confirm, safe mode does NOT fix the issue.

**Comment by fhardy32** [comment]
I am also having same problem

**Comment by Johndowne** [comment]
It’s so annoying - yes here too on iOS iPhone and iPad - doesn’t seem much better in browser though.

**Comment by jaggerc** [comment]
Same problem here

**Comment by chemgeek** [comment]
+1, this has been an issue for a few weeks now.

**Comment by emarvegt** [comment]
Same problem here. In a browser everything is fine, but in the app it’s frozen or freezingly slow. A reinstall of the app and updating to the latest version did not fix the problem.

**Comment by 54Comet** [comment]
Same problem here.  Companion app on iPhone 13, iphone 15 and iPad Pro are unusable.  Also unusable from safari on iOS on all three devices.  Stutters and lags.  Can’t get nodered to even try to launch from sidebar.  UI is snappy and works well in chrome browser on windows 10 machine.

**Comment by Solarflor** [comment]
It seems no one has been assigned to solve this issue.

**Comment by nicholashead** [comment]
By any chance y’all using iBeacon/have it installed?

**Comment by jaggerc** [comment]
I’m not using iBeacon. On May 26, 2024, at 9:35 AM, Nicholas Head ***@***.***> wrote:﻿
By any chance y’all using iBeacon/have it installed?

**Comment by themistymay** [comment]
> By any chance y’all using iBeacon/have it installed?

No

**Comment by Johndowne** [comment]
> By any chance y’all using iBeacon/have it installed?

No not me (had to Google it!)

**Comment by emarvegt** [comment]
No.

**Comment by silamon** [comment]
There have been reports before, more info can be found here: https://community.home-assistant.io/t/very-slow-performance-from-lovelace-dashboard/551612/22. 

If somebody can look into this and give more information, an improvement can be made.

**Comment by themistymay** [comment]
> There have been reports before, more info can be found here: https://community.home-assistant.io/t/very-slow-performance-from-lovelace-dashboard/551612.
> 
> If somebody can look into this and give more information, an improvement can be made.

I’ll confirm this is not my issue. I have a iOS device  (same model and OS version as impacted devices) that is not impacted by slowness of the exact same dashboards.

**Comment by majuss** [comment]
[2024-05-30_13.32.22+0200.logs.zip](https://github.com/home-assistant/frontend/files/15499144/2024-05-30_13.32.22%2B0200.logs.zip)
Here are the logs of an impacted iPhone 13. Problem is present since beginning of April.

**Comment by 54Comet** [comment]
> There have been reports before, more info can be found here: https://community.home-assistant.io/t/very-slow-performance-from-lovelace-dashboard/551612.
> 
> If somebody can look into this and give more information, an improvement can be made.

This tracks a different problem that users have faced prior to this year.  This bug is different.

**Comment by loopy321** [comment]
Also experiencing very slow app response on iOS 16 after updating to app v  2024.5.1 from 2023.12.

Prior to update was very fast.

**Comment by loopy321** [comment]
I have reset the app (v. 2024.5.1), deleted the server and re-added in the app and updated HA to 2024.6.3 and the problems persists.  It takes more than 1 minute to load the dashboard in the iOS app (iphoneX iOS v. 16.7.8) whereas in the 2023.12 version of the app it was very fast (~1-2 secs) on the same device.  

See logfile here: https://gist.github.com/loopy321/35bf742b2e80f7c3a929f15e47c52040

Only thing I see in the above log is a bunch of the following message repeating multiple times a second:
`[hakit-work-queue] [Environment.swift:71] init() > WebSocket: Received: event: for HARequestIdentifier(rawValue: 31)`

Any ideas on what to troubleshoot or where to look next??  Please.

**Comment by jaggerc** [comment]
I keep updating to every new version in hopes something is going to fix this.  I've uninstalled everything but essential integrations and restored from a backup but the problem is still there.  The app is essentially dead for me unless I'm willing to wait 1-2 minutes to turn something on or off.

**Comment by themistymay** [comment]
Not seeing any dev comments but a growing number of user reports here and on forum. Maybe @balloob can help triage?

**Comment by themistymay** [comment]
Forum topic link: https://community.home-assistant.io/t/app-is-extremely-slow/710599

**Comment by Solarflor** [comment]
@balloob , @frenck ,
Hi guys, 
I believe the solution of this problem should be a priority a (vast) majority of HA user have HA running on IPhone.
Any support is welcome.
Thanks

**Comment by nicholashead** [comment]
I will say what helped me was removing the iBeacon Tracker integration and making sure all it's entities were removed. That seemed to really speed up the app for me. I wasn't even using that integration, but it keeps adding entities nonstop as it sees them.

So maybe the slowness is directly related to the number of entities/etc you have?

**Comment by majuss** [comment]
I have 1450 entities and everything is slow as hell. 
Somebody should open an issue inside the correct repository: https://github.com/home-assistant/iOS

**Comment by carsaig** [comment]
Same issue here. The iBeacon integration is not the issue - at least not in my case.
As soon as I dare to scroll or press somewhere, the app crashes instantly. Unusable currently. 
I don't have the time currently to look into logs, sorry. Someone else needs to sort this out.

**Comment by Ivan86NL** [comment]
For a +1; Here also having the same problem for several months now.

**Comment by Solarflor** [comment]
> I think I know, what the issue is

but why only on IPhone

**Comment by carsaig** [comment]
> > I think I know, what the issue is
> 
> but why only on IPhone

Good question. No idea, whether the apps are OS-specific or whether the OS handle the datastream differently...
There is no use in speculating from my side. Someone needs to debug what is invoking the lag on an iPhone, reed some logs, etc. and possibly come up with a long-term solution.
The issue can be approached from various sides: looking at the data stream and handling between app & platform or just ignore the lack of stream management of the app and focus on the root cause on platform side and deal with the app inconsistencies later.
There has been done a lot on DB optimization on the HA core through recent updates, which is good news - but looking at the tables and entities structure...I am by no means a DB expert - it partially looks problematic the way string patterns are designed and written. At least I would assume this invokes problems downstream and requires an alternative approach or optimization efforts because some of these patterns were designed years back where devices just didn't flood systems with heaps of JSON at high frequency. Things are changing at high pace. But that's just me looking at this from far away. Part of the issue is definitely my lack of knowledge around the HA architecture when customizing things. This platform offers vast opportunities through customization but at the same time this scales the possibilities of failure, if not designed in an appropriate way, including thorough testing :-) yes. testing. Not just debugging. Unit-testing and regressions ideally :-)
In other words: this platform requires quite a lot of knowledge and digging into, before working with it at scale.

As I mentioned: The fastest way forward and immediate solution to get the app working smooth again, is to limit the no. of DB requests by controlling the  required entities and/ or domains sent to the recorder with a filter setting. This cuts the amount of DB clutter by roughly 80-90%.
Next step would be to dig deeper into the actual root cause.
In my specific case the cause for DB clutter was an nmap wifi tracker picking up too many mac addresses rotated regularly by mobile devices, despite filter settings (I was just not concise enough, using exclude rules instead of include), furthermore the bluetooth ble tracker flooding my db with known_devices due to same reason, and lastly badly designed sensors flooding the DB with state_changes every few seconds by default. Especially related to the bluetooth devices. And that scaled the amount of related/ connected data in the DB which was requested by the app. Probably the payload just exceeded some limit somewhere on the route to the App UI and crashed it.
I am re-writing my sensors currently, learning better coding patterns and practice.
I queried all the heavy-hitters in the DB but just for reference to understand where the problems arise and will proceed with inclusion rules on the recorder level, gradually re-adding domains and entities as required.
As a last step I'll off-load long-term statistics to influxDB.

**Comment by Solarflor** [comment]
@carsaig one step per time.
how I can confirm my DB is too crowded?

**Comment by carsaig** [comment]
> @carsaig one step per time. how I can confirm my DB is too crowded?
1. Go to development tools in the HA UI --> Template and paste this to get an overview of entities in use:

`{% for d in states | groupby('domain') %}
  {% if loop.first %} Domains: {{loop.length}}
  {% endif %} - {{ d[0] }} ({{ states[d[0]] | count }})
      - {{ states[d[0]] |map(attribute='entity_id')| list|join('\n      - ') }}
{% endfor %}`

2. Install two very helpful add-on's if you prefer UI-driven tools instead of CLI and raw SQL queries:
[dbstats](https://github.com/jehy/hass-addons/tree/master/dbstats)
[sqlite web](https://github.com/hassio-addons/addon-sqlite-web)
3. Look into dbstats and identify the heavy-hitters
4. start creating a list of include entities for your recorder filter

)

**Comment by Solarflor** [comment]
> 2\. dbstats

I have installed it but don't know how to config it
![image](https://github.com/home-assistant/frontend/assets/100423265/ff1c8bfa-9566-44ed-9623-dae9495ee4c6)

**Comment by carsaig** [comment]
> > 2. dbstats
> 
> I have installed it but don't know how to config it ![image](https://private-user-images.githubusercontent.com/100423265/342347195-ff1c8bfa-9566-44ed-9623-dae9495ee4c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkyMzQ1NDksIm5iZiI6MTcxOTIzNDI0OSwicGF0aCI6Ii8xMDA0MjMyNjUvMzQyMzQ3MTk1LWZmMWM4YmZhLTk1NjYtNDRlZC05NjIzLWRhZTk0OTVlZTRjNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNFQxMzA0MDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZDlhMmFlODA5Mzg3ZjM0MjFhMzI5OWNkZGY1N2Y0YjczYjllY2MxNDEwYzI3ZGI5MDlmNDZjNGExYzdlYjM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.reHZ-Md1JljXVOjMb3wDRAMsSEw0p5UYhsNyiVK2s8Y)

There is nothing to configure. Just install the add-on, let it show up in your side-pane and you're good to go.

**Comment by Solarflor** [comment]
> There is nothing to configure. Just install the add-on

Not able to start the add-on. My HA OS is running on a NAS Virtual Machine.
![image](https://github.com/home-assistant/frontend/assets/100423265/4b8940da-19df-4804-b879-854556f22685)

**Comment by Solarflor** [comment]
> 2\. sqlite web

Installed and it seems to run well.
How I can use it to have usefull information?

**Comment by majuss** [comment]
Guys, this is the wrong direction! The database is not the problem.
The same HA instance is lagging on iPhone 13 and 15 for me but Mi 10 and 12 have not a single problem.
So why are you focused on the HA server @carsaig @Solarflor ? Please just open an issue inside the correct repository, this is the wrong one!

**Comment by Solarflor** [comment]
@majuss this is why I opened this issue here but it seems nobody have interested to splve this problem

**Comment by majuss** [comment]
Why did you opened the issue here and not in the ios repository? https://github.com/home-assistant/iOS

**Comment by carsaig** [comment]
> Guys, this is the wrong direction! The database is not the problem. The same HA instance is lagging on iPhone 13 and 15 for me but Mi 10 and 12 have not a single problem. So why are you focused on the HA server @carsaig @Solarflor ? Please just open an issue inside the correct repository, this is the wrong one!

Simply because I wasn't aware the other issue existed. Or in other words: I didn't look for it because I just stumbled in here and was in a hurry. Sorry for that. Didn't mean to mash things up.

**Comment by themistymay** [comment]
Created issue: https://github.com/home-assistant/iOS/issues/2822

**Comment by loopy321** [comment]
FYI, I managed to get some significant improvement by starting over with my lovelace dashboards.  I have not isolated which card was causing the problem, but load times improved drastically.  If you are at a loss for what to try, this is my suggestion.

I went into the "raw configuration editor" copied everything to notepad and starting adding items back in.

It seemed that culprit was one of `custom:weather-chart-card` `custom:mini-graph-card` `custom:apexcharts-card` or my file sensor.videos:
```
      - type: custom:gallery-card
        entities:
          - sensor.videos
```

FWIW.

**Comment by beasthouse-au** [comment]
I managed to fix this by taking control of the default dashboard and saving it with nothing on it. I had also removed the custom cards like @loopy321 mentioned. Now I am getting fast speeds on the App and via the Safari Browser on iOS again.

**Comment by Solarflor** [comment]
> managed to fix this by taking control of the default dashboard

How? opening a new one as default dashboard?

**Comment by beasthouse-au** [comment]
> > managed to fix this by taking control of the default dashboard
> 
> How? opening a new one as default dashboard?

I just opened the default dashboard, clicked edit and it comes up with the option to take control, which I did and also selected the start with empty dashboard option.

**Comment by Solarflor** [comment]
There should be something wrong in my lovelace setting 
How your have set it in your configuration file?

This is my setting

```
lovelace:
  mode: storage
```

**Comment by beasthouse-au** [comment]
> There should be something wrong in my lovelace setting How your have set it in your configuration file?
> 
> This is my setting
> 
> ```
> lovelace:
>   mode: storage
> ```

same as this but what i meant was doing this.

![CleanShot 2024-07-02 at 00 06 12@2x](https://github.com/home-assistant/frontend/assets/78465928/fb07d424-f931-421c-bc15-f3e047a41791)


![CleanShot 2024-07-02 at 00 06 54@2x](https://github.com/home-assistant/frontend/assets/78465928/a499a307-6718-4c87-a67f-5cde085fff5e)


![CleanShot 2024-07-02 at 00 07 24@2x](https://github.com/home-assistant/frontend/assets/78465928/e95e4d19-b22f-4d6a-8199-c91e02b94deb)

**Comment by themistymay** [comment]
I'll reiterate the issue not being dashboard specific.
I have two matching phones, one is impacted and one is not so it is not a
dashboard/lovelace issue. It would have to be something underneath that is
instantiating either non-deterministically or impacted by something local
to the phone that is outside the front end and app cache.

**Comment by Solarflor** [comment]
> same as this but what i meant was doing this.

I don't have it and I don't know why.

**Comment by Solarflor** [comment]
### Solved!
I followed  @loopy321 and @beasthouse-au way and I have creating a new default dashboard with nothing on it.

**Comment by thoduh** [comment]
Since I activated Homeassistant Cloud today, the app has been working without any delay.

**Comment by beasthouse-au** [comment]
> Since I activated Homeassistant Cloud (Test month) today, the app has been working without any delay.

Yeah it's been fairly stable lately

**Comment by thoduh** [comment]
Only lasted a short time, problems are back... :-(

**Comment by rodrigofragadf** [comment]
It had improved for a while but the problem came back.

**Comment by le-clu** [comment]
For me, the latest iOS update solved it. So either it is a coincidence or the bug with Safari mentioned above was indeed fixed by Apple.

**Comment by TgF2011NL** [comment]
I have (had) the same problem. iOS 16.7.10 on iPhone X. I worked backwards and found out that the problem started after the Core 2024.9.0 update. I’ve just restored to the previous update and the iOS companion app is working smoothly as before. Companion app is the latest version. So I guess that in my case the problem is more related to the core update. On my laptop everything was working fine, even after the 2024.9.0 core update.

Just installed the core 2024.9.1 update and the problem was right back, now back to core 2024.8.3

I also tried avoiding the companion app and tried to access Home Assistant via Chrome browser on my phone: same issue, my dashboard is not loading, or not loading completly, or takes a lot of time to load.

**Comment by rodrigofragadf** [comment]
I'm struggling with slow performance in the Home Assistant app for iOS for months now, and it is currently unusable. Trying to figure out the cause, I created a test installation to gradually add my devices and integrations. The first one I added was SmartThinQ LGE Sensors, which controls my two air conditioners. I had already noticed on my main Home Assistant setup that the control cards for the air conditioners were no longer appearing. I thought it might be some kind of dashboard integration issue, but I realized that when I go to the integrations menu, select SmartThinQ LGE Sensors, choose one of the devices, and click on the climate entity, the control card doesn't appear, and neither do any sensors.

So, I installed this as the first integration in the clean test HA, and the problem persists. As soon as I click on the climate entity, or if there's an air conditioner control card on the dashboard, the HA app becomes sluggish, making it impossible to navigate the menus or even click on anything. I have to close the app and reopen it. Additionally, as soon as I click on the climate entity, the following error is shown in the log. I have another HA instance running in my other house, which integrates my air conditioners via MQTT, and this issue does not occur.

I’m unsure if this is an integration issue or a problem with HA itself, but I hadn’t noticed this log before, so I think it might be a good place to start investigating. I hope someone from the team can look into this error, as I’ve been unable to use the Home Assistant app on my phone for months, and it’s very frustrating.

Uncaught error from WebKit 605.1.15 on iOS 15.1 Script error. null @:0:0
Uncaught error from WebKit 605.1.15 on iOS 15.1 RangeError: Maximum call stack size exceeded. f (src/scoped-custom-element-registry.js:328:40) value (node_modules/element-internals-polyfill/dist/index.js:676:33) Object (constraint-validation.ts:242:33) privateSyncValidity (constraint-validation.ts:205:11) requestUpdate (delegate.ts:127:13) this (src/scoped-custom-element-registry.js:342:31) this (src/scoped-custom-element-registry.js:342:31) name (delegate.ts:101:27) value (text-field.ts:518:4) this (src/scoped-custom-element-registry.js:330:40)
Uncaught error from WebKit 605.1.15 on iOS 15.1 RangeError: Maximum call stack size exceeded. value (constraint-validation.ts:204:6) requestUpdate (delegate.ts:127:13) this (src/scoped-custom-element-registry.js:342:31) this (src/scoped-custom-element-registry.js:342:31) name (delegate.ts:101:27) value (text-field.ts:518:4) this (src/scoped-custom-element-registry.js:330:40) this (src/scoped-custom-element-registry.js:329:28) value (node_modules/element-internals-polyfill/dist/index.js:676:33) Object (constraint-validation.ts:242:33)
Uncaught error from WebKit 605.1.15 on iOS 15.1 RangeError: Maximum call stack size exceeded. FormAssociatedElement (form-associated.ts:275:6) value (constraint-validation.ts:204:6) requestUpdate (delegate.ts:127:13) this (src/scoped-custom-element-registry.js:342:31) this (src/scoped-custom-element-registry.js:342:31) name (delegate.ts:101:27) value (text-field.ts:518:4) this (src/scoped-custom-element-registry.js:330:40) this (src/scoped-custom-element-registry.js:329:28) value (node_modules/element-internals-polyfill/dist/index.js:676:33)
Uncaught error from WebKit 605.1.15 on iOS 15.1 RangeError: Maximum call stack size exceeded. value (form-associated.ts:275:6) value (constraint-validation.ts:204:6) requestUpdate (delegate.ts:127:13) this (src/scoped-custom-element-registry.js:342:31) this (src/scoped-custom-element-registry.js:342:31) name (delegate.ts:101:27) value (text-field.ts:518:4) this (src/scoped-custom-element-registry.js:330:40) this (src/scoped-custom-element-registry.js:329:28) value (node_modules/element-internals-polyfill/dist/index.js:676:33)

**Comment by Solarflor** [comment]
Hi all,
the only solution I found to survive is creating a new default dashboard (actually empty) and set it as "predefined".
In this way when you open HA you will enter in the new default dashboard without any problem.

**Comment by beasthouse-au** [comment]
> Hi all, the only solution I found to survive is creating a new default dashboard (actually empty) and set it as "predefined". In this way when you open HA you will enter in the new default dashboard without any problem.

Yep, this was the solution for me - has been fine since I did this

**Comment by TgF2011NL** [comment]
> Hi all, the only solution I found to survive is creating a new default dashboard (actually empty) and set it as "predefined". In this way when you open HA you will enter in the new default dashboard without any problem.

That's a lot of work then. I have a perfectly operating dashboard, with all kinds of sub pages and al lot of cards on them. I do an update to Core 2024.9 and only on my iPhone that dashboard stops working and to solve that I have to start all over? Looks a bit too much work.

**Comment by rodrigofragadf** [comment]
Doing more tests, I believe I’ve discovered what’s causing the issue. I copied my entire current dashboard and created a new one with all its content. I started checking which cards and entities were not showing up and removed entity by entity until the cards started to reappear.

At first, I thought it might be some problem with temperatures since all temperature cards and entities were not shown and caused freezing, but then I realized it also happened with entities showing percentages, folder sizes, energy consumption, and anything that had numbers with commas (in Brazil, we use commas instead of periods to separate decimal places).

I removed all entities displaying numbers with commas, and everything started working well again, without freezing.

Then, I did another test using Node-RED: I took the values coming from HA’s temperature sensors and created a new sensor for HA inside Node-RED. If I set the Device Class as temperature or any other class that works with commas, the issue returns, with the entity not being displayed and, depending on the number of entities with commas, causing freezing.

On the other hand, if I leave the Device Class empty, the sensor is displayed without any issues, even with the comma.

In summary, I believe this is the root cause of the problem. If you could test to see if this behavior also occurs for you, we could confirm that this is indeed the reason.

**Comment by rodrigofragadf** [comment]
After doing some more tests, I discovered that, in my case, changing the language to English in the User Options menu solved the problems. However, when using Portuguese (BR) with any number format, the problem reappears. Is anyone facing this issue using the English language?

**Comment by Solarflor** [comment]
> After doing some more tests, I discovered that, in my case, changing the language to English in the User Options menu solved the problems. However, when using Portuguese (BR) with any number format, the problem reappears. Is anyone facing this issue using the English language?

not for me

**Comment by thoduh** [comment]
> Nachdem ich einige weitere Tests durchgeführt hatte, stellte ich fest, dass in meinem Fall die Änderung der Sprache in Englisch im Menü Benutzeroptionen die Probleme löste. Bei Verwendung von Portugiesisch (BR) mit einem beliebigen Zahlenformat tritt das Problem jedoch erneut auf. Ist jemand mit diesem Problem konfrontiert, der die englische Sprache verwendet?



> After doing some more tests, I discovered that, in my case, changing the language to English in the User Options menu solved the problems. However, when using Portuguese (BR) with any number format, the problem reappears. Is anyone facing this issue using the English language?

not for me

**Comment by rodrigofragadf** [comment]
After switching to English, go back to the page where the problem occurs and try refreshing it. Open and close the app, or even try clearing the app’s cache by going to Settings -> Companion App -> Debugging -> Clear frontend cache. 

In addition to changing the language to English, try also changing the number format to 1234567.89 and run some tests. 

Try creating a new dashboard and add only the card where you identified the issue. Remove entities that contain decimal numbers and check if the card displays correctly. This worked for me both in my production instance and in my test instance, and I was even able to reproduce the error in safe mode.

**Comment by thoduh** [comment]
It's a shame that the IOS app is no longer usable. It would be really great if someone could take care of the problem.

**Comment by emarvegt** [comment]
For me, the issue was solved after deleting a massive load of unused entities which were auto-generated by the RFLink 443 MHz integration that was set to autodiscovery for a long time.
These entities were not on any dashboard that I configured but apparently bogged down the app so much that it became unusable. With the entities removed, the app is snappy and responsive. In any browser, this was never an issue.

**Comment by thoduh** [comment]
The app for the Mac is also no longer usable :-(

**Comment by chheiss** [comment]
Same problem here. Safari and other kiosk browsers load snappy fast, but iOS HA app is so slow it is basically unusable.

**Comment by ldorfmueller** [comment]
Same Problem here. Suddenly Companion App is unusable slow. I can't even get into the App-Settings anymore. Reinstall did not help. iPhone 13 Pro with iOS 18.0.1.

**Comment by ptr727** [comment]
I just tried the app while on 5G only, I wanted to pull my hair out, seems like one of those old school apps that does network IO on a UI thread; tap, wait wait wait, UI moves, tap wait wait wait, give up. On wifi at home things seem fine, but remote, unusable.

**Comment by madz12345** [comment]
I have the same problem. Iphone 14 with the latest version of IOS and applications.
